fe2-android/app/src/main/java/com/github/dedis/popstellar/model/Copyable.java

```diff
@@ -0,0 +1,47 @@
+package com.github.dedis.popstellar.model;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Simple interface describing objects that can be copied
+ *
+ * <p>It also store utility functions that can be used to create deep copy of collections
+ *
+ * @param <T> of the copied object
+ */
+public interface Copyable<T> {
+
+  /**
+   * Create a deep copy of the object
+   *
+   * @return The copy
+   */
+  T copy();
+
+  /**
+   * Create a copy of a map
+   *
+   * @param source map
+   * @param <T> type of key
+   * @param <C> type of value, must by a Copyable
+   * @return the copy
+   */
+  static <T, C extends Copyable<C>> Map<T, C> copy(Map<T, C> source) {
+    return source.entrySet().stream()
+        .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().copy()));
+  }
+
+  /**
+   * Create a copy of a map containing lists
+   *
+   * @param source map
+   * @param <E> type of key
+   * @param <T> type of value inside the list
+   * @return the copy
+   */
+  static <E, T> Map<E, List<T>> copyMapOfList(Map<E, List<T>> source) {
+    return source.entrySet().stream()
+        .collect(Collectors.toMap(Map.Entry::getKey, entry -> new ArrayList<>(entry.getValue())));
```

Comment:
don't we need `T` to be `T extends Copyable<T>` here? Don't we need to deep-copy the list contents ?

Reply:
That's a good idea especially for later update by students unfamiliar with the issue

Reply:
Actually, I thought about it and I think creating a dedicated type is a bit overkill for the future of the app.

Our goal is to move to a mostly functional app state, and most of the copy will be done on immutable objects.